Codecademy/client-modules
# feat(Global Header): add redirect parameter

### Overview



The monolith’s current header Log In & Sign Up links have redirect parameters, determined by [addRedirectParam](https://github.com/codecademy-engineering/Codecademy/blob/develop/webpack/assets/libs/url.ts#L62
)
so that the user gets redirected to the page they were at after logging in or signing up.

ex. anonymous user is on Learn Html course page. they click sign up in the app header. they sign up. they are taken to Learn HTML instead of to dashboard

*note: In doing this work, I discovered a bug in the log in page redirect functionality. i.e. redirect only works on sign up in production. spoke to Rebecca & created a bug ticket to fix that: https://codecademy.atlassian.net/browse/GROW-2230


To Test -

- Visit https://badger.codecademy.com/ where my [monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/22024) is deployed using the alpha build from this PR
- Navigate from different pages to the sign up and log in pages. Note the redirect param in the URL (*it may be easier to see if you do this side by side to production)
- Navigate from the catalog or a course page to the sign up page. 
- Sign up a user
- Skip all the prompts to be navigated elsewhere (don't click on a recommended course in the onboarding survey, for ex)
- See that you're navigated to the page you came from




### PR Checklist

- [x] Related to JIRA ticket: [EGG-850](https://codecademy.atlassian.net/browse/EGG-850)
